crates/flipperzero/src/dialogs/mod.rs

```diff
@@ -202,19 +203,22 @@ impl DialogMessageButton {
 
 impl<'a> DialogFileBrowserOptions<'a> {
     /// Creates a new dialog file browser options and initializes to default values.
-    pub fn new() -> Self {
+    pub fn new<'e: 'a>(extension: &'e CStr) -> Self {
```

Comment:
I'm not familiar with the `'e: 'a` syntax (I think it's saying `'e` must live for at least as long as `'a`).

What would the difference be between `'e: 'a` and `'a` on its own?

Reply:
Exactly, this is `'e outlives 'a`. In my understanding, if both lifetimes were the same, we would not be able to have `&'static extensions` without `'a` also being `'static`, although I may be wrong here

Reply:
I think this is where [variance](https://doc.rust-lang.org/reference/subtyping.html) comes into play.

Consider the [case of a `struct` with `&'a str`](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=5c33d5bcd5cc1b51a10c2606a005884e). Without variance we couldn't use `&'static str`  without also making the struct `'static`.

So I think in this case we would only need `'a`.

Reply:
Fair enough

I've replaced this with `'a` and added two doc-tests to validate appropriate and inappropriate use-cases.